### DISCUSSION TITLE
Is there a way to get tab-completions for this?

### DISCUSSION THREAD

**Discussion by NireBryce** [olets/zsh-abbr#102] [Q&A] [answered]
I can't tell if I'm missing something or it's impossible at the moment, but I can't find a way to hook this into completions.

I know it takes a lot of work, so I'm not asking for it to be added, just asking if there's a way to do it that's pre-existing.

thanks!

**Comment by olets** [comment]
Short answer: no :)

--

If you mean completion for `abbr` commands (like `abbr exp[TAB]`), contributions are welcome.

--

If you mean abbreviation completion, like `abbr helloworld="echo hello world"; hellow[TAB]` , I expect there's a solution that leverages zsh's alias completion or its function completion. You'd have something like

```shell
completion-for-abbr() {
  local abbreviation

  for abbreviation in $(abbr list-abbreviations); do
    alias $abbreviation=$abbreviation
  end
}

completion-for-abbr
```

or

```shell
completion-for-abbr() {
  local abbreviation

  for abbreviation in $(abbr list-abbreviations); do
    $abbreviation() {}
  end
}

completion-for-abbr
```

in your `.zshrc` below the line where zsh-abbr is added.

If it's possible, it would take finessing. Otherwise (and if you simply paste that code into your `.zshrc`), you'll get completion but you'll end up with the abbrevation _and_ the expansion:

```shell
% abbr myexample="echo my example"
% alias myexample=myexample
% myexam[Trigger completion, select `myexample`][SPACE] # % myexam echo my example
```

**Reply by NireBryce** [reply]
alright, I'll tinker with this and if I find a better way I'll get back to you.  Thanks!

I meant the abbreviations, but if I'm already dealing with this I'll see if I can add completions to abbr itself.  it's probably good as a warm up anyway.

my hands are full for the month but I'll add it to the stack.

**Reply by olets** [reply]
- Right after my first comment I decided to add command completions. That's released now, in v5.2.0! Sorry to take what could have been your contribution. PR #103 has the work, if you're interested.

**Reply by olets** [reply, marked as answer]
Documenting: And since https://github.com/olets/zsh-abbr/pull/128 some zsh-abbr subcommands have been able to complete abbreviations.
